File: backend-ruby/app/models/phong_thi.rb
class PhongThi < ApplicationRecord
  self.table_name = 'PhongThi'
  self.primary_key = 'PhongThiID'
  has_many :phan_phong_records, class_name: "PhanPhong", foreign_key: "PhongThiID"
  has_many :xep_cho_records, class_name: "XepCho", foreign_key: "PhongThiID"
  has_many :diem_danh_records, class_name: "DiemDanh", foreign_key: "PhongThiID"
end

File: backend-csharp/Models/PhongThi.cs
using System.Text.Json.Serialization;

namespace backend_csharp.Models;

public class PhongThi
{
    public int PhongThiID { get; set; }
    public string MaPhong { get; set; } = string.Empty;
    public string TenPhong { get; set; } = string.Empty;
    public string? ToaNha { get; set; }
    public int? Tang { get; set; }
    public int SucChua { get; set; }
    public int? SoHang { get; set; }
    public int? SoCot { get; set; }
    public bool TrangThai { get; set; } = true;
    public DateTime TaoLuc { get; set; }
    public DateTime? CapNhatLuc { get; set; }
    [JsonIgnore] public ICollection<PhanPhong> PhanPhongRecords { get; set; } = [];
    [JsonIgnore] public ICollection<XepCho> XepChoRecords { get; set; } = [];
    [JsonIgnore] public ICollection<DiemDanh> DiemDanhRecords { get; set; } = [];
}
